kpi/utils/query_parser/query_parser.py

```diff
@@ -114,6 +115,14 @@ def _get_value(_field, _elements):
 
         if elements[0].text == '':
             value = _get_value('', elements)
+
+            # As discussed here: https://github.com/kobotoolbox/kpi/pull/2830
+            # there strain on the server for small search queries without a
+            # specified field. The user will receive an empty list in response
+            # until using 3 or more characters
+            if len(value) < 3:
+                raise BadDefaultSearchException()
+
```

Comment:
According to be mentioned before, we could specify the message over here instead.

```python
if len(value) < 3:
    default_detail = _('The search term must be 3 or more characters') 
```

Besides, I think we discussed about using a constant (instead of hard-coding `3`) at top the class. I may be wrong, though.